src/serialize/window.js

```diff
@@ -124,6 +125,8 @@ export class ProxyWindow {
     constructor({ send, win, serializedWindow } : {| win? : CrossDomainWindowType, serializedWindow? : SerializedWindowType, send : SendType |}) {
         this.actualWindowPromise = new ZalgoPromise();
         this.serializedWindow = serializedWindow || getSerializedWindow(this.actualWindowPromise, { send });
+        this.serializedWindowType = this.serializedWindow.getType();
```

Comment:
Sorry for nitpick. But can we do this up here:

https://github.com/krakenjs/post-robot/blob/master/src/serialize/window.js#L44

`let windowType = ...`

That way we're always checking the window type when we have a full window reference. Here, `this.serializedWindow.getType()` might result in a post-message which is less optimal (and also asynchronous)

Reply:
Also can we keep the `throw` in `getType` when `windowType` is not defined, and the window is closed? I think we still have to cover that case, in case someone passes in a closed window then tries to get the type.